s3/s3_transport/include/s3_transport_logging_category.hpp

```diff
@@ -0,0 +1,34 @@
+#ifndef S3_TRANSPORT_LOGGING_CATEGORY_HPP
+#define S3_TRANSPORT_LOGGING_CATEGORY_HPP
+
+#include "irods/irods_logger.hpp"
+
+// 1. Declare the custom category tag.
+//    This structure does not need to define a body.
+//    This tag allows the logger to locate data specific to the new category.
+struct s3_transport_logging_category;
+
+// 2. Specialize the logger configuration for the new category.
+//    This also defines the default configuration for the new category.
+namespace irods::experimental
+{
+    template <>
+    class log::logger_config<s3_transport_logging_category>
+    {
+        // This defines the name that will appear in the log under the "log_category" key.
+        // The "log_category" key defines where the message originated. Try to use a name
+        // that makes it easy for administrators to determine what produced the message.
+        static constexpr const char* name = "s3_transport_logging_category";
+
+        // This is the current log level for the category. This also represents the initial
+        // log level. Use the "set_level()" function to adjust the level.
+        static inline log::level level = log::level::info;
```

Comment:
The log level in the plugin logger is explicitly declared as a public member. Is there a reason it is not here?

Reply:
I needed read access to it for the debug logging loop to print the L1desc table.  I didn't need this for the transport one.

Should I instead change it back to private with a getter message?

Reply:
Changed the plugin logger's level back to private and added a getter.

```
        // This is the current log level for the category. This also represents the initial
        // log level. Use the "set_level()" function to adjust the level.
        static inline log::level level = log::level::info;

    public:

        static auto get_level() -> decltype(level)
        {   
            return level;
        }  
```

Reply:
We know the type of `level` here. Why not use `log::level` instead of `decltype(level)`?

Reply:
Done